search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt

```diff
@@ -309,6 +318,130 @@ class AttributeInstanceService(
+                SELECT id 
+                FROM temporal_entity_attribute 
+                WHERE entity_id = :entity_id 
+                AND attribute_name = :attribute_name
+            )
+            AND instance_id = :instance_id
+            """.trimIndent()
+
+        return databaseClient
+            .sql(modifiedQuery)
+            .bind("entity_id", entityId)
+            .bind("attribute_name", attributeName)
+            .bind("instance_id", instanceId)
+            .oneToResult(
+                ResourceNotFoundException(
+                    attributeOrInstanceNotFoundMessage(attributeName, instanceId.toString())
+                )
+            ) {
+                Pair(
+                    it["temporal_entity_attribute"] as UUID,
+                    JsonUtils.deserializeObject(toJsonString(it["payload"])) as Map<String, List<Any>>
+                )
+            }
+    }
+
+    @Transactional
+    suspend fun modifyAttributeInstance(
+        entityId: URI,
+        attributeName: ExpandedTerm,
+        instanceId: URI,
+        expandedPayload: Map<String, List<Any>>
+    ): Either<APIException, Unit> = either {
+        val (teaUUID, oldPayload) = retrieveTeaUUIDAndPayload(entityId, attributeName, instanceId).bind()
+        val payload = expandedPayload.toMutableMap()
+        oldPayload.forEach {
+            payload.putIfAbsent(it.key, it.value)
+        }
+        val measuredValue = valueToDoubleOrNull(
+            (expandedPayload[NGSILD_PROPERTY_VALUE] as List<Map<String, String>>).first()[JSONLD_VALUE_KW]!!
+        )!!
+        val newInstance = AttributeInstance(
+            time = getPropertyValueFromMapAsDateTime(expandedPayload, NGSILD_OBSERVED_AT_PROPERTY)!!,
+            measuredValue = measuredValue,
+            payload = payload,
+            instanceId = instanceId,
+            temporalEntityAttribute = teaUUID,
+            timeProperty = AttributeInstance.TemporalProperty.OBSERVED_AT
+
+        )
+        deleteInstance(entityId, attributeName, instanceId).bind()
+        create(newInstance).bind()
+        updateInstanceAudit(
+            entityId,
+            attributeName,
+            instanceId,
+            measuredValue,
+            newInstance.payload
+        ).bind()
+
+        val newInstanceId = generateRandomInstanceId()
+        create(
+            newInstance.copy(
+                timeProperty = AttributeInstance.TemporalProperty.MODIFIED_AT,
+                time = ZonedDateTime.now(ZoneOffset.UTC),
+                instanceId = newInstanceId,
+                payload = Json.of(
+                    JsonUtils.serializeObject(
+                        payload
+                            .minus(NGSILD_INSTANCE_ID_PROPERTY)
+                            .plus(NGSILD_INSTANCE_ID_PROPERTY to buildNonReifiedProperty(newInstanceId.toString()))
+                    )
+                )
+            )
+        ).bind()
+    }
+
+    @Transactional
+    suspend fun updateInstanceAudit(
```

Comment:
I really don't understand this one. Why are you updating the `attribute_instance_audit` with the new value and payload?

Reply:
I modify `attribute_instance_audit` beacause I need to modify the payload and the value for the creation instance. I do it for the observedAt before and now for the createdAt.